graphql_compiler/macros/__init__.py

```diff
@@ -167,16 +170,26 @@ def get_schema_for_macro_definition(schema):
     """Returns a schema with macro directives.
 
     This returned schema can be used to validate macro definitions, and support GraphQL
-    macro editors, enabling them to autocomplete one the @macro_edge_definition and
-    @macro_edge_target directives.
+    macro editors, enabling them to autocomplete on the @macro_edge_definition and
+    @macro_edge_target directives. Some directives that are disallowed in macro edge definitions,
+    like @output and @output_source, will be removed from the directives list.
 
     Args:
         schema: GraphQLSchema over which we want to write macros
 
     Returns:
         GraphQLSchema usable for writing macros
     """
-    raise NotImplementedError()  # TODO(bojanserafimov): Implement
+    macro_definition_schema = copy(schema)
+    macro_definition_schema_directives_set = set(macro_definition_schema.get_directives()).union(
+        DIRECTIVES_REQUIRED_IN_MACRO_EDGE_DEFINITION)
+    # Exclude disallowed directives
+    macro_definition_schema_directives_set &= DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINITION
```

Comment:
I think it's better to validate that all of the current schema directives are allowed and raise a validation error if one of them isn't. (Look for the exception that fits best). Then I would add a test that validation works and update the docstring with a "Raises" block.

Reply:
If you would continue with your current approach I would separate this into functions: one to remove the unsupported directives and another to add the required directives.

Reply:
Agree with Pedro on raising an exception. The macro system is not a macro system for any graphql schema, but specifically a macro system for this graphql compiler. If it's being used with anything else, it should not promise to work and pretend that everything is ok.

Reply:
Added schema.py/_check_for_nondefault_macro_directives() to detect this issue.